backend/src/main/java/gov/cdc/usds/simplereport/idp/authentication/LiveOktaAuthentication.java

```diff
@@ -1,87 +1,86 @@
+import org.openapitools.client.model.SmsUserFactorProfile;
+import org.openapitools.client.model.UpdateUserRequest;
+import org.openapitools.client.model.User;
+import org.openapitools.client.model.UserCredentials;
+import org.openapitools.client.model.UserFactor;
+import org.openapitools.client.model.UserStatus;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Profile;
 import org.springframework.http.HttpEntity;
 import org.springframework.http.HttpHeaders;
-import org.springframework.http.HttpMethod;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
-import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
-import org.springframework.web.client.HttpClientErrorException;
 import org.springframework.web.client.RestClientException;
 import org.springframework.web.client.RestTemplate;
 
 /**
  * Created by emmastephenson on 4/28/21
  *
- * <p>Handles all Okta-related user authenticaton.
+ * <p>Handles all Okta-related user authentication.
  *
  * <p>NOTE: If you alter this file, you may need to re-record the WireMock stubs. See
  * https://github.com/CDCgov/prime-simplereport/issues/1848.
  */
 @Profile("!" + BeanProfiles.NO_OKTA_AUTH)
 @Service
 public class LiveOktaAuthentication implements OktaAuthentication {
-  private static final String USER_API_ENDPOINT = "/api/v1/users/";
   private static final String ACTIVATION_KEY = "activation";
+  private final String apiToken;
+  private final String orgUrl;
+  private final RestTemplate restTemplate;
+  private final UserApi userApi;
+  private final UserFactorApi userFactorApi;
 
-  private Client _client;
-  private String _apiToken;
-  private String _orgUrl;
-  private RestTemplate _restTemplate;
-
-  @Autowired
-  public LiveOktaAuthentication(OktaClientProperties oktaClientProperties) {
-    initialize(oktaClientProperties.getOrgUrl(), oktaClientProperties.getToken());
-  }
-
-  public LiveOktaAuthentication(String orgUrl, String token) {
-    initialize(orgUrl, token);
-  }
-
-  private void initialize(String orgUrl, String token) {
-    _client =
+  public LiveOktaAuthentication(
+      @Value("${okta.client.org-url}") String orgUrl, @Value("${okta.client.token}") String token) {
+    ApiClient client =
         Clients.builder()
             .setOrgUrl(orgUrl)
             .setClientCredentials(new TokenClientCredentials(token))
             .build();
-    _apiToken = token;
-    _orgUrl = orgUrl;
-    _restTemplate = new RestTemplate();
+    this.apiToken = token;
+    this.orgUrl = orgUrl;
+    this.restTemplate = new RestTemplate();
+    this.userApi = new UserApi(client);
+    this.userFactorApi = new UserFactorApi(client);
   }
 
   /**
    * Fetches the user account status. Does NOT correspond to the user's Okta account status, as we
    * also care about things like factor activation status.
    */
+  @Override
```

Comment:
just to understand this bit: we need `@Override` here in the upbrade because we're switching from the monolithic client API (which used to exist in the base `OktaAuthentication` class and therefore could just be inherited) to the individual endpoint ones, meaning we need to replace the base implementation with the user API client. Anything I'm missing / incorrect about?

Reply:
The changes to add `@Override` were just to clarify in the code that we are overriding this method from `OktaAuthentication` interface. This is not a necessary change at all and was just a recommendation from IntelliJ I thought may be helpful.